modules/tasksUI.js

```diff
@@ -18,104 +21,111 @@ export default class TasksUI {
-      }); */
       tasksDbInstance
-        .filter((task) => task.completed !== true)
+        // .filter((task) => task.completed !== true)
         .forEach((task) => {
           // console.log(task);
           const taskContainer = document.createElement('div');
-          taskContainer.className = 'list-element';
-          taskContainer.id = 'task-container';
+          taskContainer.className = 'list-element flex';
+          taskContainer.id = `${task.id}`;
           tasksUIelement.appendChild(taskContainer);
           const taskToogle = document.createElement('input');
           taskToogle.type = 'checkbox';
           taskToogle.className = 'task';
-          taskToogle.id = `${task.index}`;
+          taskToogle.checked = false;
+          taskToogle.id = `toogle${task.id}`;
           taskContainer.appendChild(taskToogle);
           const taskTitle = document.createElement('input');
           taskTitle.className = 'task-title-text';
-          taskTitle.id = `${task.index}`;
+          if (task.completed) {
+            taskToogle.checked = true;
+            taskTitle.style = 'text-decoration-line: line-through;';
+          } else {
+            taskToogle.checked = false;
+            taskTitle.style = '';
+          }
+          taskTitle.id = `title${task.id}`;
           taskTitle.value = task.description;
           taskContainer.appendChild(taskTitle);
 
-          taskTitle.addEventListener('keypress', (event) => {
-            if (event.key === 'Enter') {
+          taskTitle.addEventListener('change', (event) => {
+            if (taskTitle.value !== task.description) {
               event.preventDefault();
               // console.log('pressed Enter');
               this.editTask(event);
             }
           });
-          taskToogle.addEventListener('change', this.removeTask);
+
+          taskToogle.addEventListener('change', (event) => {
+            // console.log(event);
+            // console.log(event.target.id);
+            // const order = event.target.id - 1
+            const dBInstance = JSON.parse(localStorage.getItem('tasksDB'));
+            // console.log(dBInstance);
+            const elementID = document.querySelector(`#${event.target.id}`)
+              .parentElement.id;
+            const taskElement = dBInstance.find(
+              (elem) => elem.id === elementID,
+            );
+            // console.log(taskElement);
+            if (taskElement.completed) {
+              // console.log('checking task is complete');
+              taskToogle.checked = false;
+              taskElement.completed = false;
+              localStorage.setItem('tasksDB', JSON.stringify(dBInstance));
+              TasksUI.loadTasksUI();
+            } else {
+              taskToogle.checked = true;
+              taskElement.completed = true;
+              localStorage.setItem('tasksDB', JSON.stringify(dBInstance));
+              TasksUI.loadTasksUI();
+            }
+          });
```

Comment:
- [OPTIONAL]  KIndly make sure to remove all comments in keep your codebase clean.